### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR I impleented the below:

- Copied `calculate.js` and `operate.js` into a logic/ directory.
- Analyzed `calculate.js` and `operate.js`, to understand what they do.
-  added [big.js](https://www.npmjs.com/package/big-js) as dependency.
- Imported the files in `Calculator.js` component.
- Implemented the event handlers needed to use the math logic from `calculate.js` and `operate.js`.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import operate from './operate';
+
```
Comment (mateo951) [inline src/logic/calculate.js:2]: `calculate.js` and `operate.js` where added correctly ✔

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,45 +1,83 @@
 import React from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
```
Comment (mateo951) [inline src/components/Calculator.js:3]: Proper importation of calculate.js to Calculator component 📝

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,45 +1,83 @@
...
-    this.state = {};
+  constructor() {
+    super();
+    this.state = { total: '0', next: '', operation: '' };
+  }
+
+  operationsHandler(name) {
+    const object = this.state;
+    const result = calculate(this.state, name);
+    this.setState({ ...object, ...result });
```
Comment (mateo951) [inline src/components/Calculator.js:14]: Props are passed to child components of the `Calculate.js` component✔

File: src/components/Calculator.js [hunk 4]
```diff
@@ -1,45 +1,83 @@
...
-        <button type="button">AC</button>
-        <button type="button">+/-</button>
-        <button type="button">%</button>
-        <button type="button" className="operation">
+        <div className="span-4">
+          {object.total}
+          {object.operation}
+          {object.next}
+        </div>
+        <button type="button" onClick={() => this.operationsHandler('AC')}>
```
Comment (mateo951) [inline src/components/Calculator.js:26]: Implemented the click event for all buttons 🚀

**Review by mateo951 — APPROVED** [review]
### Hi @EidHachem, 

Your app is working perfectly! Well done implementing the expected events to your calculator 🤓There is nothing else to say other than... it's time to merge it 🙌 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by EidHachem** [comment]
@mateo951 Thank you so much 🌹

**Pull request merged**